src/amo/pages/Guides/index.js

```diff
@@ -312,20 +311,19 @@ export class GuidesBase extends React.Component<InternalProps> {
   getGuidesSections = (
     sections: Array<SectionsType>,
   ): React.ChildrenArray<React.Node> => {
-    const { addons, clientApp, i18n, lang } = this.props;
+    const { addons, i18n } = this.props;
+
+    const splitOn = '__LINK__';
 
     return sections.map((section) => {
-      // TODO: look into having these links use the Router (vs 'a' tag).
-      // See https://github.com/mozilla/addons-frontend/issues/6787.
-      let exploreMoreLink;
-      if (lang && clientApp) {
-        exploreMoreLink = i18n.sprintf(section.exploreMore, {
-          linkStart: `<a class="Guides-section-explore-more-link" href="/${lang}/${clientApp}${
-            section.exploreUrl
-          }">`,
-          linkEnd: '</a>',
-        });
-      }
+      const localizedExploreMoreLink = i18n.sprintf(section.exploreMore, {
```

Comment:
I think this could actually be moved into `getLocalizedTextWithLinkParts` so that the consumer (this file) never needs to know the value of `splitOn`. Instead, you could pass in `i18n`, the string (in this case it's `section.exploreMore`), and the names of the start and end link variables (in this case it's `linkStart` and `linkEnd`). With all that information, `getLocalizedTextWithLinkParts` should be able to do all the work and just return the parts as it is now doing.

Reply:
thanks @bobsilverberg. So I think I mostly followed your suggestion here. Let me know if I am missing anything tho:)